File: 프로그래머스/0/120838. 모스부호 （1）/모스부호 （1）.cs
using System;

public class Solution
{
    public string solution(string letter) 
    {
        string[] morse = 
        { 
            ".-","-...","-.-.","-..",".","..-.",
            "--.","....","..",".---","-.-",".-..",
            "--","-.","---",".--.","--.-",".-.",
            "...","-","..-","...-",".--","-..-",
            "-.--","--.."
        };
        string[] ch = 
        { 
            "a","b","c","d","e","f",
            "g","h","i","j","k","l",
            "m","n","o","p","q","r",
            "s","t","u","v","w","x",
            "y","z"
        };
        
        string[] letterList = letter.Split(' ');
        
        string result = "";
        for(int i=0; i<letterList.Length; i++)
        {
            for(int j=0; j<morse.Length; j++)
            {
                if(letterList[i] == morse[j])
                {
                    result += ch[j];
                }
            }
        }
        return result;
    }
}

File: 프로그래머스/0/120838. 모스부호 （1）/모스부호 （1）.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(string letter) 
{
    map<string, string> morse = 
    {
        {".-","a"},{"-...", "b"},{"-.-.","c"},{"-..","d"},{".","e"},{"..-.","f"},
        {"--.","g"},{"....","h"},{"..","i"},{".---","j"},{"-.-","k"},{".-..","l"},
        {"--","m"},{"-.","n"},{"---","o"},{".--.","p"},{"--.-","q"},{".-.","r"},
        {"...","s"},{"-","t"},{"..-","u"},{"...-","v"},{".--","w"},{"-..-","x"},
        {"-.--","y"},{"--..","z"}
    };
    
    string str = "";
    string result = "";
    for(char &c : letter)
    {
        if(c != ' ')
        {
            str +=c;
        }
        else if(c == ' ')
        {
            result += morse[str];
            str = "";
        }
    }
    if(!str.empty())
    {
        result += morse[str];
    }
    return result;
}